nvflare/app_opt/pt/file_list_model_locator.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+
+from nvflare.apis.dxo import DXO
+from nvflare.apis.event_type import EventType
+from nvflare.apis.fl_context import FLContext
+from nvflare.app_common.abstract.model import model_learnable_to_dxo
+from nvflare.app_common.abstract.model_locator import ModelLocator
+from nvflare.app_opt.pt.file_model_persistor import PTFileModelPersistor
+
+
+class PTFileListModelLocator(ModelLocator):
+    def __init__(self, pt_persistor_id: str, model_list={}):
+        """The ModelLocator's job is to find and locate the models inventory saved during training.
+
+        Args:
+            pt_persistor_id (str): ModelPersistor component ID
+            model_list: a list of model_names and locations
+        """
+        super().__init__()
+
+        self.pt_persistor_id = pt_persistor_id
+
+        self.model_persistor = None
+        self.model_list = model_list
+
+    def handle_event(self, event_type: str, fl_ctx: FLContext):
+        if event_type == EventType.START_RUN:
+            self._initialize(fl_ctx)
+
+    def _initialize(self, fl_ctx: FLContext):
+        engine = fl_ctx.get_engine()
+        self.model_persistor: PTFileModelPersistor = engine.get_component(self.pt_persistor_id)
+        if self.model_persistor is None or not isinstance(self.model_persistor, PTFileModelPersistor):
+            raise ValueError(
+                f"pt_persistor_id component must be PTFileModelPersistor. " f"But got: {type(self.model_persistor)}"
+            )
+        self.model_persistor.load_model(fl_ctx)
```

Comment:
Why do we need to call "load_model" here? I understand that some persistor needs to know the model format, but that should be handled by that persistor itself (e.g. it can call load_model in its event handler).

Reply:
The purpose of this call to enforce the persistence_manager creation in the PTFileModelPersistor. Changed it to use lazy initialization there.